bookwyrm/timezone_middleware.py

```diff
@@ -0,0 +1,17 @@
+import pytz
+
+from django.utils import timezone
+
+
+class TimezoneMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if request.user.is_authenticated:
+            timezone.activate(pytz.timezone(request.user.preferred_timezone))
+        else:
+            timezone.activate(pytz.utc)
+        response = self.get_response(request)
+        timezone.deactivate()
+        return response
```

Comment:
This should be a Django-friendly timezone middleware https://docs.djangoproject.com/en/3.1/topics/i18n/timezones/

I am not 100% sure if the explicit call to deactivate is necessary. What I read led me to believe that that is performed after the view is called for the request and response, and I thought it might be safe to clean up. https://docs.djangoproject.com/en/3.1/topics/http/middleware/#writing-your-own-middleware

Even if Django recycles threads I think that this might be superfluous, though.